lib/Ast.ml

```diff
@@ -50,7 +50,7 @@ type 'e expression =
   | Variable of identifier
   | IntNumeral of string
   | RealNumeral of string
-  | FunApp of identifier * 'e list
+  | FunApp of Mir.funKind * identifier * 'e list
```

Comment:
I think we also need to do NRFunApp :(

Reply:
I don't know why, but I didn't think users could define side-effecting functions. I can't seem to work out how this is being handled at the moment in `Stan_math_code_gen.ml` - do you have any pointers?